packages/cf-util-http-file/src/saveFile.js

```diff
@@ -1,22 +1,14 @@
 export default function saveFile(filename, file, mimeType) {
-  mimeType = mimeType || 'data:application/octet-stream';
-  // Data URIs cannot be used for navigation, for scripting, or to
-  // populate frame or iframe elements in IE. Instead for IE specifically,
-  // use `msSaveBlob` to save the file.
+  mimeType = mimeType || 'application/octet-stream';
   if ('msSaveBlob' in global.navigator) {
     global.navigator.msSaveBlob(new Blob([file]), filename);
     return;
   }
 
-  // Create a new anchor tag, set its href attribute to be a data url
-  // comprised of the contents of the response (aka our file), then click
-  // the link. This will prompt the browser to download the file.
+  const fileBlob = new Blob([file], { type: mimeType });
   const link = global.document.createElement('a');
-  link.href = mimeType + ',' + global.encodeURI(file);
+  link.href = global.URL.createObjectURL(fileBlob);
   if (filename) link.download = filename;
 
-  // Dispatch fake click event to download the file
-  const event = global.document.createEvent('MouseEvents');
-  event.initEvent('click', true, true);
-  link.dispatchEvent(event);
+  link.click();
```

Comment:
How sure are you that this works across all of our browsers? I noticed that MDN states `click()` will not initiate navigation on anchor tags https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click so it follows that it may not trigger a download.